packages/x-date-pickers/src/LocalizationProvider/LocalizationProvider.tsx

```diff
@@ -116,4 +142,19 @@ LocalizationProvider.propTypes = {
    * Locale for the date library you are using
    */
   locale: PropTypes.oneOfType([PropTypes.object, PropTypes.string]),
+  /**
+   * Locale for components texts
+   */
+  localeText: PropTypes.shape({
```

Comment:
On the datagrid we don't generate the shape (I don't know why because `localeText` is not in the blacklist of `generateProptypes` (cc @m4theushw)
We should probably have `PropTypes.object` here as well

Reply:
Maybe add `localeText` to the black list then